models/quality_measures/intermediate/nqf0053_osteoporosis_management_in_women_who_had_a_fracture/quality_measures__int_nqf0053_exclusions.sql

```diff
@@ -0,0 +1,311 @@
+
+)
+
+, nonacute_outpatient_counts as (
+
+    select
+          patient_id
+        , exclusion_type
+        , count(distinct exclusion_date) as encounter_count
+    from nonacute_outpatient_advanced_illness
+    group by patient_id, exclusion_type
+
+)
+
+, valid_advanced_illness_exclusions as (
+
+    select
+          acute_inpatient_advanced_illness.patient_id
+        , acute_inpatient_advanced_illness.exclusion_date
+        , acute_inpatient_advanced_illness.exclusion_reason
+        , acute_inpatient_advanced_illness.exclusion_type
+    from acute_inpatient_advanced_illness
+    left join acute_inpatient_counts
+      on acute_inpatient_advanced_illness.patient_id = acute_inpatient_counts.patient_id
+    where acute_inpatient_counts.encounter_count >= 1
+
+    union all
+
+    select
+        nonacute_outpatient_advanced_illness.patient_id
+      , nonacute_outpatient_advanced_illness.exclusion_date
+      , nonacute_outpatient_advanced_illness.exclusion_reason
+      , nonacute_outpatient_advanced_illness.exclusion_type
+    from nonacute_outpatient_advanced_illness
+    left join nonacute_outpatient_counts
+      on nonacute_outpatient_advanced_illness.patient_id = nonacute_outpatient_counts.patient_id
+    where nonacute_outpatient_counts.encounter_count >= 2
+
+
+)
+-- advanced illness end
+
+, frailty_patients_within_defined_window as (
+
+    select
+          frailty_within_defined_window.patient_id
+        , frailty_within_defined_window.exclusion_date
+        , frailty_within_defined_window.exclusion_reason
+        , 'measure specific exclusion for defined window' as exclusion_type
+    from frailty_within_defined_window
+    
+)
+
+, exclusions as (
+
+    select * from valid_advanced_illness_exclusions
+
+    union all
+
+    select * from valid_dementia_exclusions
+
+    union all
+
+    select * from valid_institutional_snp
+
+    union all
+
+    select * from valid_hospice_palliative
+
+    union all
+
+    select * from exclusion_procedure_and_medication
+
+    union all
+
+    select * from frailty_patients_within_defined_window
+
+)
+
+
+, combined_exclusions as (
+
+  select 
+      exclusions.*
+    , denominator.age
+  from exclusions
+  inner join {{ref('quality_measures__int_nqf0053_denominator')}} as denominator
+      on exclusions.patient_id = denominator.patient_id
+
+)
+
+, valid_exclusions as (
+
+    select 
+        * 
+    from combined_exclusions
+    where exclusion_type = 'institutional_snp'
+    and age >= 66
+
+    union all 
+
+    select 
+        *
+    from combined_exclusions
+    where exclusion_type in ('advanced_illness', 'dementia')
+    and age between 66 and 80
+
+    union all 
+
+    select 
+      *
+    from combined_exclusions
+    where exclusion_type = 'measure specific exclusion for defined window'
+    and age >= 81
+
+    union all
+
+    select 
+        *
+    from combined_exclusions
+    where exclusion_type not in ('institutional_snp', 'advanced_illness', 'dementia')
```

Comment:
The exclusion type `measure specific exclusion for defined window` needs to be included in this list. The way it's written now is allowing patients under 81 with frailty to be allowed into the exclusions, but they don't meet the age requirement.